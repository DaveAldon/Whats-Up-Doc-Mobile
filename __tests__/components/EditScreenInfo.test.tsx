import React from "react";
import renderer from "react-test-renderer";
import EditScreenInfo from "../../components/EditScreenInfo";

describe("<EditScreenInfo />", () => {
  const mock = { path: "path" };

  beforeEach(() => {
    jest.useFakeTimers();
  });

  it("has children", () => {
    const tree = renderer.create(<EditScreenInfo {...mock} />).toJSON();
    expect(tree?.children.length).toBeGreaterThan(0);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<EditScreenInfo {...mock} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
