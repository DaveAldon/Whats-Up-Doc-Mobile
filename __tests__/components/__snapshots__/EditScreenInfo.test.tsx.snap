// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditScreenInfo /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "marginHorizontal": 50,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(0,0,0,0.8)",
          },
          Object {
            "fontSize": 17,
            "lineHeight": 24,
            "textAlign": "center",
          },
        ]
      }
    >
      Open up the code for this screen:
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(0,0,0,0.05)",
          },
          Array [
            Object {
              "borderRadius": 3,
              "paddingHorizontal": 4,
            },
            Object {
              "marginVertical": 7,
            },
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
            },
            Array [
              undefined,
              Object {
                "fontFamily": "space-mono",
              },
            ],
          ]
        }
      >
        path
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "rgba(0,0,0,0.8)",
          },
          Object {
            "fontSize": 17,
            "lineHeight": 24,
            "textAlign": "center",
          },
        ]
      }
    >
      Change any of the text, save the file, and your app will automatically update.
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "marginHorizontal": 20,
          "marginTop": 15,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingVertical": 15,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2f95dc",
            },
            Object {
              "textAlign": "center",
            },
          ]
        }
      >
        Tap here if your app doesn't automatically update after making changes
      </Text>
    </View>
  </View>
</View>
`;
